src/lua/error.lua

```diff
@@ -143,6 +144,14 @@ local function error_unpack(err)
         local f = fields[i]
         result[ffi.string(f._name)] = mp_decode(f._data)
     end
+    -- Hide redundant fields from unpack output (gh-9101).
+    if compat.box_error_unpack_type_and_code.current == 'new' then
```

Comment:
I would use `:is_new()`, because `.curent` may be `'default'`. See PR #8858.

Reply:
Fixed.

```diff
diff --git a/src/lua/error.lua b/src/lua/error.lua
index 01efbb96d..90e18a282 100644
--- a/src/lua/error.lua
+++ b/src/lua/error.lua
@@ -145,7 +145,7 @@ local function error_unpack(err)
         result[ffi.string(f._name)] = mp_decode(f._data)
     end
     -- Hide redundant fields from unpack output (gh-9101).
-    if compat.box_error_unpack_type_and_code.current == 'new' then
+    if compat.box_error_unpack_type_and_code:is_new() then
         result.custom_type = nil
         result.base_type = nil
         if result.code == 0 then
```

Reply:
This one is missing as well